src/server.ts

```diff
@@ -16,11 +17,23 @@ router.get("/nodes/:id", (ctx) => {
   ctx.body = getNodeAndChildren(Number(ctx.params.id))
 })
 
+router.post("/nodes", (ctx) => {
+  const body = ctx.request.body
+  ctx.body = storeNewNode(body.name, body.type, Number(body.parent))
```

Comment:
```suggestion
  ctx.body = storeNewNode(body.name, body.type, Number(body.parentId))
```

Reply:
Shit, how do you create a node inside the root? `Number(body.parentId)` seems to convert `null` into `0` :P

Reply:
No need for type casting if you we are sending json body. a number is a number. if it's null it's null and that's that :P